src/poetry/console/commands/check.py

```diff
@@ -58,6 +71,18 @@ def validate_classifiers(
         return errors, warnings
 
     def handle(self) -> int:
+        if self.option("lock"):
+            if self.poetry.locker.is_locked() and self.poetry.locker.is_fresh():
+                self.line("poetry.lock is consistent with pyproject.toml.")
+                return 0
+            self.line_error(
+                "<error>"
+                "Error: poetry.lock is not consistent with pyproject.toml. "
+                "Run `poetry lock [--no-update]` to fix it."
+                "</error>"
```

Comment:
I just noticed that we print this error even if there is no lockfile. Maybe, we should make a distinction between "no lockfile" and "inconsistent lockfile"?

Reply:
Likely resolved by 6125b6a18cdbb7aa1ef8386fcbbe461f3ac81d52 since those errors are now separate checks